### PR TITLE
fix: [CDS-50937]: gitops source changes

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/utils/utils.ts [hunk 1]
```diff
@@ -150,6 +150,11 @@ export function isMultiTypeExpression(type: MultiTypeInputType): boolean {
   return type === MultiTypeInputType.EXPRESSION
 }
 
+export function getLastURLPathParam(URL: string): string {
+  const splitArray = URL.split('/')
+  return splitArray.length > 0 ? splitArray[splitArray.length - 1] : ''
+}
+
```
Comment (ishantmehtaharness) [inline src/modules/10-common/utils/utils.ts:157]: Do we need this in common utils?

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx [hunk 2]
```diff
@@ -21,10 +23,12 @@ import {
   Layout,
   Select,
   SelectOption,
-  Text
+  Text,
+  useToaster
 } from '@harness/uicore'
 import { Color, FontVariation, PopoverProps } from '@harness/design-system'
 import cx from 'classnames'
+import * as Yup from 'yup'
```
Comment (ishantmehtaharness) [inline src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx:31]: `import Yup from 'yup'`

File: src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx [hunk 3]
```diff
@@ -167,7 +177,7 @@ const ConfigureGitopsRef = (props: any): JSX.Element => {
           accountIdentifier: accountId,
           projectIdentifier,
           orgIdentifier,
-          identifier: identifier
+          identifier: getLastURLPathParam(defaultTo(repoURL, ''))
```
[the remark this answers is not in the document]
Comment (satyam-harness) [reply src/modules/75-cd/pages/get-started-with-cd/ConfigureGitops/ConfigureGitops.tsx:180]: yeah, making this change

File: src/modules/10-common/utils/utils.ts (hunk 1 above)
Comment (satyam-harness) [reply src/modules/10-common/utils/utils.ts:157]: may be reused somewhere else

**Review by ishantmehtaharness — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**